rpc/src/v1/impls/signing.rs

```diff
@@ -170,67 +167,78 @@ impl<C: 'static, M: 'static> ParitySigning for SigningQueueClient<C, M> where
 		let res = self.dispatch(RpcConfirmationPayload::Decrypt((address.clone(), data).into()), address.into());
 
 		let (ready, p) = futures::oneshot();
-		// TODO [todr] typed handle_dispatch
-		self.handle_dispatch(res, |response| {
-			match response {
-				Ok(RpcConfirmationResponse::Decrypt(data)) => ready.complete(Ok(data)),
-				Err(e) => ready.complete(Err(e)),
-				e => ready.complete(Err(errors::internal("Unexpected result.", e))),
-			}
-		});
-
-		p.then(|result| futures::done(result.expect("Ready is never dropped nor canceled."))).boxed()
+
+		// when dispatch is complete
+		res.then(move |res| {
+			// register callback via the oneshot sender.
+			handle_dispatch(res, move |response| {
```

Comment:
Most probably `handle_dispatch` could now return Future instead of accepting a callback (but it's a separate issue)